app/src/main/java/com/jwisozk/flightreviews/ReviewListAdapter.kt

```diff
@@ -0,0 +1,92 @@
+package com.jwisozk.flightreviews
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.annotation.NonNull
+import androidx.recyclerview.widget.RecyclerView
+import com.jwisozk.flightreviews.AbsParamCell.ViewType
+
+class ReviewListAdapter(
+    @NonNull private val inflater: LayoutInflater,
+    @NonNull private val viewModel: ParamViewModel,
+    @NonNull private val reviewListActionListener: ReviewListActionListener
+) : RecyclerView.Adapter<AbsHolder>() {
+
+    @NonNull
+    private val viewTypeValues = ViewType.values()
+    private val cells = ArrayList<AbsParamCell>()
+
+    override fun onAttachedToRecyclerView(recyclerView: RecyclerView) {
+        super.onAttachedToRecyclerView(recyclerView)
+        val labelTypeValues =
+            AbsParamCell.LabelType.values()
+                .sliceArray(IntRange(1, AbsParamCell.LabelType.values().size - 2))
+        for (position in labelTypeValues.indices) {
+            cells.add(
+                RatingBarParamCell(
+                    ViewType.RATING_BAR,
+                    labelTypeValues[position],
+                    recyclerView.context.getString(labelTypeValues[position].id),
+                    viewModel,
+                    reviewListActionListener
+                )
+            )
+        }
+        cells.add(
+            EditTextParamCell(
+                ViewType.EDIT_TEXT,
+                recyclerView.context.getString(AbsParamCell.LabelType.TEXT.id),
+                viewModel
+            )
+        )
+        cells.add(
+            ButtonParamCell(
+                ViewType.SUBMIT_BUTTON,
+                viewModel,
+                reviewListActionListener
+            )
+        )
+    }
```

Comment:
`RecyclerView.Adapter` should receive `List` of cells from the outside and just render them. The logic of building this `List` should be inside the `ViewModel`.